src/Stormpath/Provider/ProviderAccountRequest.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Stormpath\Provider;
+
+/*
+ * Copyright 2013 Stormpath, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use Stormpath\DataStore\DataStore;
+
+interface ProviderAccountRequest
+{
+    function getProviderData(DataStore $dataStore);
```

Comment:
We shouldn't need the data store on this method, just get `getProviderData` with no arguments. We might need a `ProviderAccountResolver` where we provide the data store to that class. See how is done in the [Java SDK for reference](https://github.com/stormpath/stormpath-sdk-java/blob/master/impl/src/main/java/com/stormpath/sdk/impl/application/DefaultApplication.java#L319).